wlauto/workloads/apklaunch/__init__.py

```diff
@@ -0,0 +1,77 @@
+#    Copyright 2015 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+
+from time import sleep
+
+from wlauto import Workload, Parameter
+from wlauto import File
+from wlauto.exceptions import ConfigError
+
+
+class ApkLaunchWorkload(Workload):
+    name = 'apklaunch'
+    description = '''
+    Installs and runs a .apk file, waits wait_time_seconds, and tests if the app
+    has started successfully.
+    '''
+    supported_platforms = ['android']
+
+    parameters = [
+        Parameter('apk_file', description='Name to the .apk to run'),
+        Parameter('pkg_name', mandatory=True,
+                  description='Package name of Android app in apk_file'),
+
+        Parameter('install_required', kind=bool, default=True,
+                  description='Set to false if the .apk has already been installed'),
+        Parameter('uninstall_required', kind=bool, default=False,
+                  description='Set to true if the package should be uninstalled'),
+        Parameter('wait_time_seconds', kind=int, default=3,
+                  description='Seconds to wait before testing if the app is alive')
+    ]
+
+    def validate(self):
+        if self.install_required:
+            if self.apk_file is None:
+                raise ConfigError('{}: install_required is true but apk_file not set'
+                                  .format(self.name))
+
+    def setup(self, context):
+        if self.install_required:
+            apk_file = context.resolver.get(File(self, self.apk_file))
+            self.logger.info("Installing {}".format(apk_file))
+            return self.device.install(apk_file)
+
+    def run(self, context):
+        self.logger.info("Starting {}".format(self.pkg_name))
+        self.device.execute('monkey -p {} -c android.intent.category.LAUNCHER 1'
+                            .format(self.pkg_name))
```

Comment:
Makes sense, this is just my Android noobiness showing. I'll keep the wait_time_seconds paramter, though, as apps could always crash after startup appears complete.